packages/design-system/src/organisms/formStepHeader/index.tsx

```diff
@@ -93,38 +108,61 @@ export const FormStepHeader: React.FC<FormStepHeaderProps> = ({
   steps,
   currentStep,
 }) => {
-  const listRef = useRef<FlatList>(null)
+  const shouldChangePosition = useBreakpointValue({
+    base: true,
+    md: false,
+  })
+  const [stepsWidth, setStepsWidth] = useState<Record<string, number>>({})
+
   const currentStepIndex = useMemo(
     () => steps.findIndex((step) => step.id === currentStep),
     [steps, currentStep],
   )
-  const onScrollToIndexFailed = useMemo(
-    () => makeScrollToIndexFailedHandler(listRef),
-    [listRef],
-  )
 
-  useEffect(() => {
-    listRef.current?.scrollToIndex({ index: currentStepIndex })
-  }, [currentStepIndex])
+  const updateWidth = useCallback((id: string, width: number) => {
+    setStepsWidth((prev) => ({ ...prev, [id]: width }))
+  }, [])
+
+  const position = useMemo(() => {
+    let position_ = 0
+    for (let i = 0; i < currentStepIndex; i++) {
+      position_ +=
+        stepsWidth[steps[i].id] +
+        SEPARATOR_WIDTH +
+        SEPARATOR_HORIZONTAL_MARGIN * 2
+    }
+    if (position_) {
+      position_ -= SEPARATOR_WIDTH / 2
+    }
+    return position_
+  }, [stepsWidth, currentStepIndex, steps])
 
   return (
-    <FlatList
-      ref={listRef}
-      horizontal
-      scrollEnabled={false}
-      initialScrollIndex={currentStepIndex}
-      showsHorizontalScrollIndicator={false}
-      data={steps}
-      ItemSeparatorComponent={Separator}
-      keyExtractor={(item) => item.id}
-      onScrollToIndexFailed={onScrollToIndexFailed}
-      renderItem={({ item, index }) => (
-        <Step
-          step={index + 1}
-          title={item.title}
-          checked={index <= currentStepIndex}
-        />
-      )}
-    />
+    <View style={[styles.stepsContainer]}>
+      <View
+        style={[
+          container.row,
+          container.alignCenter,
+          shouldChangePosition && {
```

Comment:
I think it'd be more clear that way

```suggestion
          isSmallScreen && {
```